### ISSUE TITLE
Material 3 Chip layout semantics are not within spec

### ISSUE THREAD

**Issue by Curstantine** [flutter/flutter#121959]
## Steps to Reproduce

Excluding, since this behavior is apparent even in the [sample app](https://flutter.github.io/samples/web/material_3_demo/#/).

## Expected results:

Material 3 compliant chip layout with 32dp height.

1. Layout height specified in the [spec](https://m3.material.io/components/chips/specs#b6d645e0-5632-4667-9ef5-8d920adc6f1d):
     
      <img width="75%" src="https://user-images.githubusercontent.com/69104500/222941386-078fc1fe-bec4-41cf-99d5-de9147164ff9.jpg" />

2. Example in Compose Material Catalog:
   
      <img  width="75%" src="https://user-images.githubusercontent.com/69104500/222941630-83f10dc5-975e-4082-a834-5931bf8b4d22.jpg" />

## Actual results:

1. [Material 3 Sample App](https://flutter.github.io/samples/web/material_3_demo/#/):

      <img  width="75%" src="https://user-images.githubusercontent.com/69104500/222941575-861c9f01-06b3-433f-93a5-373baafb7937.jpg" />

2. Widget inspection tool on `ActionChip(label: const Text("Vol 2"), onPressed: () {})`:
<img src="https://user-images.githubusercontent.com/69104500/222941616-ba522819-0fda-4e7e-a8bc-e6af7acd0a72.png" />

<details>
<summary>Code sample</summary>

```dart
import "package:flutter/material.dart";

void main() {
  runApp(const App());
}

class App extends StatelessWidget {
  const App({super.key});

  @override
  Widget build(context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: true),
      home: const Scaffold(
        body: Center(child: Chip(label: Text("Hello World"))),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
[√] Flutter (Channel stable, 3.7.3, on Microsoft Windows [Version 10.0.19045.2604], locale en-US)
    • Flutter version 3.7.3 on channel stable at C:\SDKs\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9944297138 (3 weeks ago), 2023-02-08 15:46:04 -0800
    • Engine revision 248290d6d5
    • Dart version 2.19.2
    • DevTools version 2.20.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at C:\Users\<REDACTED>\AppData\Local\Android\sdk
    • Platform android-33-ext4, build-tools 33.0.1
    • Java binary at:
      C:\Users\<REDACTED>\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\213.7172.25.2113.9123335\jre\bin\jav      a
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[√] Android Studio (version 2021.3)
    • Android Studio at
      C:\Users\<REDACTED>\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\213.7172.25.2113.9123335
    • Flutter plugin version 71.2.3
    • Dart plugin version 213.7433
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] VS Code (version 1.76.0)
    • VS Code at C:\Users\<REDACTED>\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.60.0

[√] Connected device (1 available)
    • <REDACTED> (mobile) • <REDACTED> • android-arm64 • Android 12 (API 31)

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

</details>

**Comment by exaby73** [comment]
Hello. It looks like the physical shape of the chip (from border to border) looks correct. The 48 pixel you see in the inspector is including some padding

![image](https://user-images.githubusercontent.com/50555895/223093294-c48b4afd-f757-4bcc-8f93-5d9768938cb5.png)

**Comment by Curstantine** [comment]
Indeed, however the chip according to my understanding the spec doesn't contain vertical padding.

Max height is 32dp, with vertical center and horizontal padding of 16dp each (8dp in case of an icon)

Ref:
<img width="1024" alt="image" src="https://user-images.githubusercontent.com/69104500/223096131-b614c4c9-4498-4945-971c-2a038eb5e452.png">

**Comment by exaby73** [comment]
This padding exists so that you can lay out your chips easier in a row or column. This seems to be an intentional design decision

<details>
<summary>Code Sample</summary>

```dart
class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(title)),
      body: const Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Chip(
              avatar: Icon(Icons.settings),
              label: Text('Chip 1'),
            ),
            Chip(
              avatar: Icon(Icons.settings),
              label: Text('Chip 2'),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

![image](https://user-images.githubusercontent.com/50555895/223124433-bb2722fa-f416-40f8-a226-6028c5576800.png)

**Comment by Curstantine** [comment]
Hmm, I'm not sure how inner padding could help with a row layout. Though, is there way to replicate a material 3 compliant chip? Constraining the height makes the text overflow, so replicating the exact component adds much bloat.

The padding makes it looks out of place when coupled with the other components.

Jetpack compose and MDC follows the spec without adding vertical inner padding.

Oh, and I think you misunderstood what I wanted to point out, I'm talking about the *inner* vertical padding between the outline and text.
Refer to:

1. Flutter:
![Screenshot_20230306-200156_Chrome](https://user-images.githubusercontent.com/69104500/223139836-197e652b-7825-4788-bc63-3dcc43c333dd.jpg)

2. Compose
![Screenshot_20230306-200131_Compose Material Catalog](https://user-images.githubusercontent.com/69104500/223139845-c1cf1b95-d79e-4b27-baca-836bf9a0fac1.jpg)

**Comment by TahaTesser** [comment]
The layout height is indeed higher than Compose version. I can get it to look similar by adjusting padding and visual density. 

```dart
        visualDensity: VisualDensity.comfortable,
        padding: const EdgeInsets.symmetric(horizontal: 8.0),
```

![Screenshot 2023-04-19 at 14 56 38](https://user-images.githubusercontent.com/48603081/233067957-87754bff-3661-4e1f-853f-036625ced506.png)

**Comment by wildsylvan** [comment]
can we get it to look exactly like the [m3 spec](https://m3.material.io/components/chips/specs) _by default_?
